org.mapstruct.eclipse/src/org/mapstruct/eclipse/internal/MapperAnnotationCompletionProposalComputer.java

```diff
@@ -0,0 +1,87 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.eclipse.internal;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.ui.ISharedImages;
+import org.eclipse.jdt.ui.JavaUI;
+import org.eclipse.jdt.ui.text.java.JavaContentAssistInvocationContext;
+import org.eclipse.jface.text.contentassist.CompletionProposal;
+import org.eclipse.jface.text.contentassist.ICompletionProposal;
+
+public class MapperAnnotationCompletionProposalComputer extends AbstractAnnotationCompletionProposalComputer {
+    private static final List<String> COMPONENT_MODEL_TYPES = Arrays.asList( "default", "cdi", "spring", "jsr330" );
+
+    @Override
+    protected List<ICompletionProposal> computeCompletionProposals(JavaContentAssistInvocationContext javaContent,
+                                                                   ICompilationUnit compilationUnit,
+                                                                   int invocationOffset, IAnnotation annotation)
+        throws JavaModelException {
+        if ( isInRange( invocationOffset, annotation.getSourceRange().getOffset(), annotation
+                                                                                             .getSourceRange()
+                                                                                             .getLength() ) ) {
+            return parseCompilationUnit(
+                compilationUnit,
+                invocationOffset,
+                String.valueOf( javaContent.getCoreContext().getToken() ) );
+        }
+        return EMPTY_PROPOSALS;
+    }
+
+    protected List<ICompletionProposal> parseCompilationUnit(ICompilationUnit compilationUnit, int invocationOffset,
+                                                             String token) {
+        final List<ICompletionProposal> returnValue = new ArrayList<ICompletionProposal>();
+
+        final Set<String> propertiesToProcess = new HashSet<String>( COMPONENT_MODEL_TYPES );
```

Comment:
Is there any need for this copy? Couldn't you iterate over `COMPONENT_MODEL_TYPES` directly (maybe make wrap it via `Collections.unmodifiableList()` at the declaration).